### PR TITLE
Use EDM for Travis CI builds

### PR DESCRIPTION
**Pull request by jwiggins**
🎶 👢 🐈 👢 🐈 👢 🐈...

### CODE REVIEW

**Comment by jwiggins** [comment]
Okie dokie then.

I'm not sure why wx is failing, but it was coming from Ubuntu before and now it's coming from EDM. I've marked it as an expected failure for now. Someone with better wx kung fu should take a look.

Having 2.7 as the environment which runs the CI script was unexpectedly painful because `click` is passing in arguments as `unicode` objects. It's all better now, but some of the code was written in anger. 🙄

**Comment by jvkersch** [comment]
Cool! Mind if I steal this for Chaco (reviewing along the way)?

**Comment by jvkersch** [comment]
Also, I wonder if it would make sense to have this EDM related code in a separate repo that can be submoduled (or installed) into other repos. Thoughts @jwiggins @corranwebster @itziakos ?

**Comment by jwiggins** [comment]
Please steal this for Chaco! And share the bugs you fix! :wink:

**Comment by corranwebster** [comment]
The code is different enough that it may not be useful to pull out the core into some other place (presumably Traits): the command-line arguments for the test runner tool here are different than those for TraitsUI, for example, and I think that is reasonable.  That makes the amount of extractable shared code fairly small, I suspect.

But I could be wrong.

File: ci/install-edm.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -e
+
+install_edm() {
+    local EDM_MAJOR_MINOR="$(echo "$INSTALL_EDM_VERSION" | sed -E -e 's/([[:digit:]]+\.[[:digit:]]+)\..*/\1/')"
```
Comment (jvkersch) [inline ci/install-edm.sh]: Maybe abort early if `INSTALL_EDM_VERSION` is not set?

File: .travis.yml [hunk 2]
```diff
@@ -1,57 +1,61 @@
...
     - libfreetype6-dev
-    - python-cairo
+
 env:
-  - ETS_TOOLKIT=wx PILLOW='pillow'
-  - ETS_TOOLKIT=qt4 PILLOW='pillow'
-  - ETS_TOOLKIT=null.image PILLOW='pillow<3.0.0'
-  - ETS_TOOLKIT=null.image PILLOW='pillow'
+  global:
+    - INSTALL_EDM_VERSION=1.5.2
```
Comment (itziakos) [inline .travis.yml]: why not 1.7.1?

Comment (jwiggins) [inline .travis.yml]: I'm not up-to-date with EDM versions. TraitsUI is using this version

Comment (itziakos) [inline .travis.yml]: Well we should fix traitsui also the current version is 1.7.1

Comment (jwiggins) [inline .travis.yml]: Agreed

File: ci/install-edm.sh (hunk 1 above)
Comment (jwiggins) [inline ci/install-edm.sh]: Among other fixes... yes.

File: .travis.yml (hunk 2 above)
Comment (jwiggins) [inline .travis.yml]: Actually... I _disagree_. 1.7.1 doesn't install correctly

**Comment by jwiggins** [comment]
Ugh... pycairo. I even tested it locally. Travis is having _none of it_

File: .travis.yml (hunk 2 above)
Comment (itziakos) [inline .travis.yml]: Can you point to the error?

Comment (jwiggins) [inline .travis.yml]: Sorry, yes: https://travis-ci.org/enthought/enable/builds/248823643

Comment (itziakos) [inline .travis.yml]: I yes, I now remember there is a bug in the `sh` installer for `1.7.1`

**Comment by mdickinson** [comment]
When trying to use this approach for local development, I'm running into issues with missing Cython. I did:
```
taniyama:enable mdickinson$ edm install -y wheel click coverage
```
followed by
```
taniyama:enable mdickinson$ edm run -- python ci/edmtool.py install --runtime=2.7 --toolkit=pyqt
```
The `install` step fails with a traceback:
```
creating build/src.macosx-10.6-x86_64-2.7/kiva/quartz
Traceback (most recent call last):
  File "setup.py", line 297, in <module>
    **config)
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/site-packages/numpy/distutils/core.py", line 169, in setup
    return old_setup(**new_attr)
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/site-packages/numpy/distutils/command/install.py", line 62, in run
    r = self.setuptools_run()
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/site-packages/numpy/distutils/command/install.py", line 56, in setuptools_run
    self.do_egg_install()
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/site-packages/setuptools/command/install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/site-packages/setuptools/command/bdist_egg.py", line 152, in run
    self.run_command("egg_info")
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/site-packages/numpy/distutils/command/egg_info.py", line 18, in run
    self.run_command("build_src")
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/site-packages/numpy/distutils/command/build_src.py", line 147, in run
    self.build_sources()
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/site-packages/numpy/distutils/command/build_src.py", line 164, in build_sources
    self.build_extension_sources(ext)
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/site-packages/numpy/distutils/command/build_src.py", line 323, in build_extension_sources
    sources = self.generate_sources(sources, ext)
  File "/Users/mdickinson/.edm/envs/enable-test-2.7-pyqt-pillow/lib/python2.7/site-packages/numpy/distutils/command/build_src.py", line 376, in generate_sources
    source = func(extension, build_dir)
  File "kiva/quartz/setup.py", line 22, in generate_c_from_cython
    from Cython.Compiler import Main
ImportError: No module named Cython.Compiler
```

**Comment by mdickinson** [comment]
Ah, I see. `Cython` is needed on OS X, but not on Linux.

**Comment by jwiggins** [comment]
Thanks for the input @mdickinson! I think I made the mistake of not cleaning before testing on macOS. Cython has been (conditionally) added now.

**Comment by jwiggins** [comment]
If there are no further comments, I'd like to merge this soon

**Comment by mdickinson** [comment]
SGTM. I was very confused for a while: I couldn't see the commit where Cython was added. I'm guessing that there was a force push at some point?

**Comment by jwiggins** [comment]
Yeah, sorry. Using GitLab has taught me some anti-social git habits (`git commit --amend` and `git rebase -i`)

**Pull request merged**